server/index.js

```diff
@@ -22,18 +22,113 @@ app.get('/:id', (req, res) => {
 });
 
 app.get('/api/experiences/:id', (req, res) => {
-  const experienceId = parseInt(req.params.id, 10);
+  const expId = {
+    experienceId: parseInt(req.params.id, 10),
+  };
 
-  db.getExperiences(experienceId)
-    .then((data) => {
+  db.getExperiences(expId, (err, data) => {
+    if (err) {
+      res.status(400).send({
+        message: err.message,
+      });
+    } else {
+      res.status(200).send(data);
+    }
+  });
+});
+
+app.post('/api/experiences/', (req, res) => {
+  console.log('post here');
+  const experienceId = parseInt(req.body.experienceId, 10);
+  const averageRating = parseInt(req.body.averageRating, 10);
+  const numberOfReviews = parseInt(req.body.numberOfReviews, 10);
+  const duration = parseInt(req.body.duration, 10);
+  const groupSize = parseInt(req.body.groupSize, 10);
+  const costPerPerson = parseInt(req.body.costPerPerson, 10);
+  const newExp = {
+    experienceId,
+    title: req.body.title,
+    city: req.body.city,
+    state: req.body.state,
+    country: req.body.country,
+    category: req.body.category,
+    activity: req.body.state,
+    averageRating,
+    numberOfReviews,
+    duration,
+    groupSize,
+    includes: req.body.includes,
+    cuisine: req.body.cuisine,
+    hostedLanguages: req.body.hostedLanguages,
+    costPerPerson,
+    imageUrls: req.body.imageUrls,
+    videoUrl: req.body.videoUrl,
+  };
+  console.log('newExp server', newExp);
+
+  db.createExperience(newExp, (err, data) => {
+    if (err) {
+      res.status(400).send({
+        message: err.message,
+      });
+    } else {
       res.status(200).send(data);
-    })
-    .catch((err) => {
+    }
+  });
+});
+
+app.put('/api/experiences/:id', (req, res) => {
+  const experienceId = parseInt(req.body.experienceId, 10);
+  const averageRating = parseInt(req.body.averageRating, 10);
+  const numberOfReviews = parseInt(req.body.numberOfReviews, 10);
+  const duration = parseInt(req.body.duration, 10);
+  const groupSize = parseInt(req.body.groupSize, 10);
+  const costPerPerson = parseInt(req.body.costPerPerson, 10);
+  const newExp = {
+    experienceId,
+    title: req.body.title,
+    city: req.body.city,
+    state: req.body.state,
+    country: req.body.country,
+    category: req.body.category,
+    activity: req.body.state,
+    averageRating,
+    numberOfReviews,
+    duration,
+    groupSize,
+    includes: req.body.includes,
+    cuisine: req.body.cuisine,
+    hostedLanguages: req.body.hostedLanguages,
+    costPerPerson,
+    imageUrls: req.body.imageUrls,
+    videoUrl: req.body.videoUrl,
+  };
```

Comment:
Maybe this code could be factored out and used for both put and post?

Reply:
I agree. Thanks!